Cleanups

SeanFeldman opened the pull request:
- object initializers
- redundant type specifiers
- unnecessary public modifiers
- unnecessary null initialization for objects
- unnecessary parentheses
- unnecessary array type specifier
- unused variables
- nameof instead of hard-coded names
- unused namespace usings
- unneeded cast for enum values
- unnecessary boxing for nunit
- prefer async version of methods over sync

![image](https://cloud.githubusercontent.com/assets/1309622/22363497/fa1c03d8-e427-11e6-9f51-5dcdb6ea6c12.png)

msftclas commented:
Hi __@SeanFeldman__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

vinaysurya commented on src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs:
```diff
@@ -16,26 +16,26 @@ namespace Microsoft.Azure.ServiceBus.Amqp
-        public const string TimeSpanName = AmqpConstants.Vendor + ":timespan";
-        public const string UriName = AmqpConstants.Vendor + ":uri";
-        public const string DateTimeOffsetName = AmqpConstants.Vendor + ":datetime-offset";
+        private const string EnqueuedTimeUtcName = "x-opt-enqueued-time";
```
>private [](start = 8, length = 7)

nit: Remove explicit private modifier.. We have typically tried to keep this practice through our solution.

SeanFeldman replied:
👍 
BTW, `LockTokenName, `PrefilteredMessageHeadersName`, and `PrefilteredMessagePropertiesName` are never used...

vinaysurya commented on src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs:
```diff
@@ -139,7 +139,7 @@ internal AmqpMessageSender(string entityName, MessagingEntityType entityType, Se
             request.Map[ManagementConstants.Properties.SequenceNumbers] = new[] { sequenceNumber };
 
-            var response = await this.ExecuteRequestResponseAsync(request);
+            await this.ExecuteRequestResponseAsync(request);
```
>await this.ExecuteRequestResponseAsync(request); [](start = 12, length = 48)

we should handle a response that is not OK and throw exception here.

SeanFeldman replied:
left a `TODO` comment

vinaysurya commented on src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs:
```diff
@@ -81,10 +81,7 @@ protected override Task OnCloseAsync()
-                    await
-                        ((AmqpMessageReceiver)this.InnerReceiver).ExecuteRequestResponseAsync(amqpRequestMessage)
-                            .ConfigureAwait(false);
+                await ((AmqpMessageReceiver)this.InnerReceiver).ExecuteRequestResponseAsync(amqpRequestMessage).ConfigureAwait(false);
```
>await [](start = 16, length = 5)

Same here. we should handle Not OK case and throw with the correct exception here

SeanFeldman replied:
same TODO

vinaysurya commented on src/Microsoft.Azure.ServiceBus/Primitives/MessagingEntityType.cs:
```diff
@@ -10,6 +10,6 @@ public enum MessagingEntityType
         Subscriber = 2,
         Filter = 3,
         Namespace = 4,
```
>Namespace = 4, [](start = 8, length = 14)

could remove this

SeanFeldman replied:
"this"? 😉

vinaysurya commented:
Thanks Sean for doing this. Have some minor comments, could you please take a look and recreate PR

jtaubensee commented:
Recreate? You just mean push a new commit, right?

vinaysurya commented:
Yes

On Jan 27, 2017 10:51 AM, "John Taubensee" <notifications@github.com> wrote:

> Recreate? You just mean push a new commit, right?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/azure-service-bus-dotnet/pull/80#issuecomment-275743155>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AT_DQolaTC0kr3QmWCHmIIDnZj-Vs-EEks5rWjyggaJpZM4Lvf7e>
> .
>

SeanFeldman commented:
I think you both meant "incorporate the changes" 😛 
done.

vinaysurya commented on src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs:
```diff
@@ -139,7 +139,13 @@ internal AmqpMessageSender(string entityName, MessagingEntityType entityType, Se
+            // TODO: no idea what code and exception should be thrown, but the should be
+            if (response.StatusCode != AmqpResponseStatusCode.OK)
+            {
+                // throw
```
@SeanFeldman , can you please merge with the latest pack and then throw 
throw response.ToMessagingContractException();

SeanFeldman replied:
>  latest pack

@vinaysurya I'm not up to speed with the team\`s jargon - you mean `dev` branch?

SeanFeldman commented:
@vinaysurya I had to merge conflicts and since no idea if the tests are green or not as it takes forever to run them :( Let me know if something got screwed in the process. Thanks.

vinaysurya commented:
@SeanFeldman, looks good to me. Not sure where the rest of the comments got lost, but regarding your comment about unused constants in LockTokenName, PrefilteredHeaderName etc in AmqpMessageConverter.cs, yes lets take them off too

SeanFeldman commented:
rebased on the latest `dev` and removed unused constants from `AmqpMessageConverter.cs`.
@vinaysurya ready for your review/merge (getting tired of rebasing and resolving conflicts 😺 )

SeanFeldman commented:
Feeli clean! How about you?
![image](https://cloud.githubusercontent.com/assets/1309622/22531487/55c143ea-e89e-11e6-8443-8d92016a2505.png)
